app/controllers/enterprises_controller.rb

```diff
@@ -88,4 +108,14 @@ def reset_order_cycle(order, distributor)
   def set_noindex_meta_tag
     @noindex_meta_tag = true unless current_distributor.visible?
   end
+
+  def inject_json_ams(name, data)
+    options = {
+      each_serializer: Api::EnterpriseSerializer,
+      data: OpenFoodNetwork::EnterpriseInjectionData.new
+    }
+    serializer_instance = ActiveModel::ArraySerializer.new(data, options)
+
+    { name: name, json: serializer_instance.to_json }
+  end
```

Comment:
I prefer to have this duplicate for now and then abstract things than having it buried in InjectionHelper where it's hard to understand.